### ISSUE TITLE
Code duplication in RtTransaction

### ISSUE THREAD

**Issue by Vatavuk** [zold-io/java-api#59]
We have code duplication in `RtTransaction`. Each method parses transaction string in a same manner:
```java
final String value = new UncheckedText(
    new IoCheckedScalar<>(
        new ItemAt<>(
            index, new SplitText(this.transaction, ";")
        )
    ).value()
).asString();
```
We can extract this to a method or a class.
Furthermore we can simplify the above example to:
```java
final String ident = new IoCheckedScalar<>(
    () -> new ItemAt<>(0, new SplitText(this.transaction, ";"))
        .value().asString()
).value();
```

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@ee5f821e-71b5-4706-b3c9-61ba92e4e645, thread: pool-62-thread-1
com.zerocracy.radars.github.ThrottledComments[74] java.lang.IllegalStateException: Can't post an identical message to zold-io/java-api#59: @llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue
```

[0.28.15](https://github.com/zerocracy/farm): Issue: zold-io/java-api#59, Action: `opened`

**Comment by victornoel** [comment]
@Vatavuk yes, or we could merge it with the pattern matching that follows and match the whole string instead of first splitting it by `;`.
Check yegor256/cactoos#941 that I opened in cactoos that is related too.

**Comment by llorllale** [comment]
@Vatavuk let's wait for yegor256/cactoos#941

**Comment by edmoffo** [comment]
The blocker referenced in this thread, yegor256/cactoos#941 (the `Strict` text validator), was closed on 2019-04-06, so the wait condition no longer holds. The duplication is still present in `src/main/java/io/zold/api/RtTransaction.java`, where seven methods construct `new ItemAt<>(index, new SplitText(this.transaction, ";"))` inline at lines 110, 135, 148, 172, 192, 216, and 240. The straightforward extraction is a private helper that returns the raw `Text` at a given index and lets each caller chain the conversion it needs; the risk is that several call sites then run `asString()` followed by `Long.parseLong`, `Instant.parse`, or a regex match, so a helper that pre-coerces to `String` would force a redundant parse-and-rewrap on the typed branches. Picking the right return type up front is the only non-mechanical part of this refactor.
